### PR TITLE
Getting runtime stats from replica

### PR DESCRIPTION
**Pull request by prateekpandey14**
This will fetch the replica stats through http request

### CODE REVIEW

**Comment by kmova** [comment]
Merging this for now. 

This needs to be enhanced to take the vsm-name instead of replica IP address.

File: command/vsm_stats.go [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+	"strings"
+	"time"
+
+	"github.com/rancher/go-rancher/client"
+)
+
+type Stats struct {
+	client.Resource
+	ReplicaCounter  int64 `json:"replicacounter"`
+	RevisionCounter int64 `json:"revisioncounter"`
```
Comment (AmitKumarDas) [inline command/vsm_stats.go]: The last time I talked with @payes I felt these counters to be similar to raft indexes ?
If it is possible lets use standard conventions.

File: command/vsm_stats.go [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+	if err := flags.Parse(args); err != nil {
+		return 1
+	}
+	args = flags.Args()
+	if len(args) < 1 {
+		c.Ui.Error(c.Help())
+		return 1
+	}
+	fmt.Printf("%2s %15s\n", "RevisionCounter", "ReplicaCounter")
+	for counter := 0; counter <= 100; counter++ {
```
Comment (AmitKumarDas) [inline command/vsm_stats.go]: Need to figure out a better way to handle this ?
Will a counter and a sleep be good ?

File: command/vsm_stats.go [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+		if err != nil {
+			fmt.Println("\nERROR:", err)
+			return -1
+		}
+		fmt.Printf("\r    %-17d %d", stats.RevisionCounter, stats.ReplicaCounter)
+	}
+	return 0
+}
+
+func NewReplicaClient(address string) (*ReplicaClient, error) {
```
Comment (AmitKumarDas) [inline command/vsm_stats.go]: Lot of technical stuff are programmed here in this func
I would have provided code level comments that helps one understand the need of each.

File: command/vsm_stats.go [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"net/url"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/rancher/go-rancher/client"
```
Comment (AmitKumarDas) [inline command/vsm_stats.go]: Can this dependency be avoided ?

**Pull request merged**